tests/integration_tests/test_catboost.py

```diff
@@ -0,0 +1,154 @@
+    study.optimize(objective, n_trials=1)
+    assert study.trials[0].state == optuna.trial.TrialState.PRUNED
+
+    study = optuna.create_study(pruner=DeterministicPruner(False))
+    study.optimize(objective, n_trials=1)
+    assert study.trials[0].state == optuna.trial.TrialState.COMPLETE
+    assert study.trials[0].value == 1.0
+
+
+METRICS = ["AUC", "Accuracy"]
+VALID_NAMES = ["validation_0", "validation_1"]
+
+
+@pytest.mark.parametrize("metric", METRICS)
+@pytest.mark.parametrize("valid_name", VALID_NAMES)
+def test_catboost_pruning_callback_init_param(metric: str, valid_name: str) -> None:
+    def objective(trial: optuna.trial.Trial) -> float:
+
+        train_x = np.asarray([[1.0], [2.0]])
+        train_y = np.asarray([[1.0], [0.0]])
+        valid_x = np.asarray([[1.0], [2.0]])
+        valid_y = np.asarray([[1.0], [0.0]])
+
+        pruning_callback = CatBoostPruningCallback(trial, metric, valid_name)
+        param = {
+            "objective": "Logloss",
+            "eval_metric": metric,
+        }
+
+        gbm = cb.CatBoostClassifier(**param)
+        gbm.fit(
+            train_x,
+            train_y,
+            eval_set=[(valid_x, valid_y), (valid_x, valid_y)],
+            verbose=0,
+            callbacks=[pruning_callback],
+        )
+
+        # evoke pruning manually.
+        pruning_callback.check_pruned()
+
+        return 1.0
+
+    study = optuna.create_study(pruner=DeterministicPruner(True))
+    study.optimize(objective, n_trials=1)
+    assert study.trials[0].state == optuna.trial.TrialState.PRUNED
+
+    study = optuna.create_study(pruner=DeterministicPruner(False))
+    study.optimize(objective, n_trials=1)
+    assert study.trials[0].state == optuna.trial.TrialState.COMPLETE
+    assert study.trials[0].value == 1.0
+
+
+@pytest.mark.parametrize(
+    "metric, valid_name",
+    [
+        ("foo_metric", "validation"),
+        ("AUC", "foo_valid"),
+    ],
+)
+def test_catboost_pruning_callback_errors(metric: str, valid_name: str) -> None:
+    def objective(trial: optuna.trial.Trial) -> float:
+
+        train_x = np.asarray([[1.0], [2.0]])
+        train_y = np.asarray([[1.0], [0.0]])
+        valid_x = np.asarray([[1.0], [2.0]])
+        valid_y = np.asarray([[1.0], [0.0]])
+
+        pruning_callback = CatBoostPruningCallback(trial, metric, valid_name)
+        param = {
+            "objective": "Logloss",
+            "eval_metric": "AUC",
+        }
+
+        gbm = cb.CatBoostClassifier(**param)
+        gbm.fit(
+            train_x,
+            train_y,
+            eval_set=[(valid_x, valid_y)],
+            verbose=0,
+            callbacks=[pruning_callback],
+        )
+
+        # evoke pruning manually.
+        pruning_callback.check_pruned()
+
+        return 1.0
+
+    # Unknown validation name or metric
+    study = optuna.create_study(pruner=DeterministicPruner(False))
+    # catboost terminates with a SystemError.
+    with pytest.raises(SystemError):
```

Comment:
In my understanding, `ValueError` is raised by the catboostcallback, so is `ValueError` more specific than `SystemError`?
```suggestion
    with pytest.raises(ValueError):
```

Reply:
ValueError is raised by the catboostcallback, then SystemError is raised by catboost.
This is how it is currently implemented. Do you have a better idea ?

Reply:
Ah, indeed. Thank you for your explanation!